File: How To Reverse Address/PHP/WhitePages-AddressLookup/views/location.php
<div class="detail_box">
    <h1>Location</h1>
    <div class="detail_boxin">
        <p>
            <?php echo $value['location']['address_line1']; ?>
            <br />
            <?php echo $value['location']['city'] .' '. $value['location']['state_code'] .' '. $value['location']['postal_code']; ?>
        </p>
    </div>
    <div class="detail_boxin">
        <p><span>Receiving Mail:</span> <?php echo $value['location']['is_receiving_mail']; ?></p>
        <?php
        if (!empty($value['location']['usage'])) { ?>
            <p><span>Usage:</span>  <?php echo $value['location']['usage']; ?></p>
        <?php
        }
        ?>
        <?php
        if (!empty($value['location']['delivery_point'])) { ?>
            <p><span>Delivery Point:</span>
                <?php
                if (!empty($value['location']['delivery_point'])) {
                    $deliveryPieces = explode("Unit", $value['location']['delivery_point']);
                    echo $deliveryPieces[0] . " Unit";
                    ?>
                <?php
                }
                ?>
            </p>
        <?php
        }
        ?>
    </div>
</div>

File: whitePages-graph/app/models/location.rb
class Location

  attr_accessor :entity, :city, :postal_code, :state_code,  :zip4, :standard_address_line1

  def initialize(entity)
    @entity = entity
    @city = city
    @postal_code = postal_code
    @state_code = state_code
    @standard_address_line1 = standard_address_line1
    @zip4 = zip4
  end

  def data
    {
        :city => city,
        :postal_code => postal_code,
        :state_code => state_code,
        :zip4 => zip4,
        :standard_address_line1 => standard_address_line1
    }
  end


  private

  def city
    entity['city']
  end

  def postal_code
    entity['postal_code']
  end

  def zip4
    entity['zip4']
  end

  def state_code
    entity['state_code']
  end

  def standard_address_line1
    entity['standard_address_line1']
  end
end
